get_file_copy_test.go

```diff
@@ -0,0 +1,84 @@
+package getter
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"testing"
+	"time"
+)
+
+// OneDoneContext is a context that is
+// cancelled after a first done is called.
+type OneDoneContext bool
+
+var _ context.Context = new(OneDoneContext)
```

Comment:
I actually had to research this as I have never seen this pattern. It seems like an interface assertion "test", although it passes off the test to the compiler, not the actual testing framework.

Given that this is a test-only context implementation and the type assertions are naturally handled through other fields here that take a context, this is not necessary, as discussed offline.

FWIW, it's probably better these are moved to an explicit test, such as:

```go
func TestSomeTypeImplementsSomeInterface(t *testing.T) {
       var i interface{} = new(SomeType)
       if _, ok := i.(SomeInterface); !ok {
               t.Fatalf("expected %t to implement SomeInterface", i)
       }
}
```

Source: https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables

Reply:
This is a pretty common pattern in our codebases so I'd say its okay! (the `var _ INTERFACETYPE = CONCRETETYPE` compile-time assertion).

Reply:
I didn't know that!

Given that the code naturally asserts `context.Context`, I still don't know if it's 100% necessary, but I've updated the comment to remove the alternate patterns. Good for me to know too for the future! 🙂 

@azr if there's a reason why this might be necessary, please err on the side of your judgement. Otherwise this LGTM, so feel free to merge after you've reviewed.

Reply:
I first did that before writing the cancellation test so that the compiler would tell me what and where I did wrong, but now that this struct is used as a `context.Context` in the test this is pretty much unnecessary; I agree.

I removed it.